pkg/controller/risingwave_controller.go

```diff
@@ -355,17 +355,18 @@ func (c *RisingWaveController) reactiveWorkflow(risingwaveManger *object.RisingW
 			if err != nil {
 				if apierrors.IsNotFound(err) {
 					l.V(1).Info("Not found, Unlock")
-					lockIdx = append(lockIdx, i)
+					continue
 				} else {
 					l.Error(err, "Failed to get risingwavescaleview")
 					return ctrlkit.RequeueIfErrorAndWrap("unable to get risingwavescaleview", err)
 				}
 			} else if s.Name == scaleView.Name && s.UID != scaleView.UID {
-				lockIdx = append(lockIdx, i)
+				continue
 			}
+			aliveScaleView = append(aliveScaleView, s)
```

Comment:
Use `DeepCopy` to avoid unexpected updates on shallowed copied fields like arrays and maps.

```suggestion
			aliveScaleView = append(aliveScaleView, *s.DeepCopy())
```